packages/compass-editor/src/json-editor.tsx

```diff
@@ -838,11 +844,65 @@ const BaseEditor = React.forwardRef<EditorRef, EditorProps>(function BaseEditor(
         readOnlyExtension,
         EditorView.updateListener.of((update) => {
           updateEditorContentHeight();
+          const editorText = editor.state.sliceDoc() ?? '';
+          if (update.focusChanged && editorViewRef.current?.hasFocus) {
+            if (editorText.trim() === '') {
+              const [updatedQuery, cursorPosition] =
+                lenientlyFixQuery(editorText);
+
+              void sheduleDispatch(editorViewRef.current, {
+                changes: {
+                  from: 0,
+                  to: editorViewRef.current.state.doc.length,
+                  insert: updatedQuery,
+                },
+                selection: EditorSelection.cursor(cursorPosition || 0),
+              });
+            }
+          }
+
           if (update.docChanged) {
-            const text = editor.state.sliceDoc() ?? '';
-            onChangeTextRef.current?.(text, update);
+            onChangeTextRef.current?.(editorText, update);
           }
         }),
+        EditorView.domEventHandlers({
+          paste(event, view) {
```

Comment:
Same as above, this doesn't belong here, this behavior should be scoped to the query option editor. If native `paste` event is not propagated from editor component like the blur / focus ones are (I haven't checked), we can use this place to add handling for `onPaste` callback prop here.

If existing editor ref methods are not enough (I am still pretty sure that what you're doing is just applying a snippet that looks kinda like `{<user paste>${}}`) we should add another one to cover this use case, but should be very specific what interface this opens as generally we should limit the amount of imperative operations we do with the editor